lib/pymedphys/_mosaiq/delivery.py

```diff
@@ -273,128 +264,77 @@ def delivery_data_sql(connection, field_id):
-            reference_txfieldpoint_results = test_txfieldpoint_results
-            test_txfield_results, test_txfieldpoint_results = delivery_data_sql(
-                connection, field_id
-            )
+    if len(raw_txfield_results) != 1:
+        raise ValueError(
+            f"The return results from txfield query gave {raw_txfield_results}. "
+            "Expected exactly one row."
+        )
+    meterset = np.array(raw_txfield_results[0]).astype(float)
+
+    if len(raw_txfieldpoint_results) == 0:
+        raise ValueError("No TxFieldPoints were returned.")
+
+    txfieldpoint_results = pd.DataFrame(
+        data=raw_txfieldpoint_results,
+        columns=[
+            "Index",
+            "A_Leaf_Set",
+            "B_Leaf_Set",
+            "Gantry_Ang",
+            "Coll_Ang",
+            "Coll_Y1",
+            "Coll_Y2",
+        ],
+    )
 
-    # return the txfield and point results, including rowversions
-    return test_txfield_results, test_txfieldpoint_results
+    return meterset, txfieldpoint_results
 
 
 class DeliveryMosaiq(DeliveryBase):
     @classmethod
     def from_mosaiq(cls, connection, field_id):
-        mosaiq_delivery_data = cls._from_mosaiq_base(connection, field_id)
-        reference_data = (
-            mosaiq_delivery_data.monitor_units,
-            mosaiq_delivery_data.mlc,
-            mosaiq_delivery_data.jaw,
-        )
+        total_mu, tx_field_points = delivery_data_sql(connection, field_id)
 
-        delivery_data = cls._from_mosaiq_base(connection, field_id)
-        test_data = (delivery_data.monitor_units, delivery_data.mlc, delivery_data.jaw)
-
-        agreement = False
-
-        while not agreement:
-            agreements = []
-            for ref, test in zip(reference_data, test_data):
-                agreements.append(np.all(ref == test))
-
-            agreement = np.all(agreements)
-            if not agreement:
-                print("Converted Mosaiq delivery data was conflicting.")
-                print(
-                    "MU agreement: {}\nMLC agreement: {}\n"
-                    "Jaw agreement: {}".format(*agreements)
-                )
-                print("Trying again...")
-                reference_data = test_data
-                delivery_data = cls._from_mosaiq_base(connection, field_id)
-                test_data = (
-                    delivery_data.monitor_units,
-                    delivery_data.mlc,
-                    delivery_data.jaw,
-                )
-
-        return delivery_data
-
-    @classmethod
-    def _from_mosaiq_base(cls, connection, field_id):
-        txfield_results, txfieldpoint_results = fetch_and_verify_mosaiq_sql(
-            connection, field_id
-        )
-
-        total_mu = np.array(txfield_results[0]).astype(float)
-        cumulative_percentage_mu = txfieldpoint_results[:, 0].astype(float)
+        cumulative_percentage_mu = tx_field_points["Index"].to_numpy(dtype=float)
```

Comment:
If this is from TxFieldPoints You need to be careful to renormalize against the last  value.  Some conventional plans can use 1.0 as the final cumulative meterset weight.  While you aren't trying to deal with Particle Therapy those can be absolute meterset or something rather arbitrary. This is probably OK for ELEKTA linacs and Monaco as the TPS, but formally, it's not guaranteed to be %

Reply:
Heya @msobolewski,

Can I give a shout-out to the value of peer review (and the value of having @sjswerdloff on board) for this one. Every time I've seen this that last point has always been 1.0, so I assumed that is how it was. However, potentially a Monaco upgrade (maybe) or a different TPS, would have completely thrown that assumption asunder and we would have been reporting total MU/cumulative MU as way off.

---

Thanks @sjswerdloff for this 🙂, massively appreciated.

Reply:
Fixed in 4da5a90a4dbd3e6d58e2660dfb7b2beeb0fa76a2